### ISSUE TITLE
Potential false positive for `no-conditional-expect`

### ISSUE THREAD

**Issue by wjhsf** [vitest-dev/eslint-plugin-vitest#834]
```ts
  const testCases = [
    { input: 'object', output: { key: 'value' } },
    { input: 'null', output: null },
  ]
  test.each(testCases)('the thing', ({ input, output }) => {
    const actual = doTheThing(input)
    const expected = output && expect.objectContaining(output)
    expect(actual).toBe(expected)
  })
```

I would expect (heh) that the goal of the `no-conditional-expect` rule is to report conditional assertions. In the code above, `expect.objectContaining` is reported as an error, even though it is conditionally used to create a value, rather than make an assertion. Is this a false positive?
